src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl service = new RegistrationServiceImpl();
+    private StorageDaoImpl storage;
+
+    @BeforeEach
+    void setUp() {
+        storage = new StorageDaoImpl();
+        service = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void registerExistingLogin_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin("TESTLOGIN");
+        storage.add(existingUser);
```

Comment:
For setting up test data, use low-level methods like `Storage.people.add()` directly instead of `service.register()` method as mentioned in the checklist.